### ISSUE TITLE
As a researcher, I want URLs I enter in metadata fields to be clickable (i.e. alternativeURL) so that it's easier to follow links

### ISSUE THREAD

**Issue by shlake** [IQSS/dataverse#3337]
The new (v4.4) **alternativeURL** metadata field should be clickable when citation metadata is displayed. 

On metadata entry, the Watermark says "Enter full URL".

Solution: In citation.tsv file the field **displayFormat** is empty. In all the other metadata fields that are "URL" fields (and are clickable) have the following as the **displayFormat**:
`<a href="#VALUE">#VALUE</a>`

So I think that is what **alternativeURL** should have:
**displayFormat**   =     `<a href="#VALUE">#VALUE</a>`

On this record you can see the non-clickable link:
https://demo.dataverse.org/dataset.xhtml?persistentId=doi:10.5072/FK2/YJMBUN

**Comment by scolapasta** [comment]
@shlake correct me if I'm wrong, but the display format for other URLs is: "<a href="#VALUE">#VALUE</a>"; not just #VALUE

Regardless, currently that displayFormat is only used for compound types. This is the first case of a URL in a primitive type. Using display format may be the correct path, bit just setting it won't actually change anything with the current logic.

**Comment by shlake** [comment]
The display format for the other URLs is: 
`
 <a href="#VALUE">#VALUE</a>`

it has the HTML code around the #VALUE. (Which I think you typed out in your comment above, but got generated as a link when saved.)

And you are correct, I was going to update the ticket on Monday - me setting the displayFormat to the above code (as the other URLs are set to), does not make it hyperlinked when displaying with other metadata (on the metadata tab).

**Comment by scolapasta** [comment]
Ah, I fell into same trap you did! (that's why I thought you only put "#VALUE" because your original comment also got generated as a link. But, OK, we are on same page now. :)

**Comment by tdilauro** [comment]
Just a note that we have run into a similar problem in our software metadata block. We have primitive types that we'd like to have render as links. We set the displayFormat to \<a href="#VALUE">#VALUE\</a>,
as mentioned above. I checked the database to verify that they had made it in properly. 

@scolapasta Is there any fundamental reason to restrict displayFormat to just the compound fields?

**Comment by pdurbin** [comment]
I'm a little confused about if there's a bug to fix or if documentation for the tsv format is lacking (#3168). Or both. 😄

**Comment by shlake** [comment]
Maybe start with updating/clarifying the documentation https://docs.google.com/document/d/1idh4wvCcoxSo7rPkolOkTD5Wvn6PQO__uXqkf7eQujA/edit#heading=h.xqa9co8gtrwh

Maybe the problem is with the "way it works" (documentation) or a bug with how things are being displayed (thinking that a field with a href value "should" be displayed as a hyperlink, and it isn't).

**Comment by scolapasta** [comment]
There is an issue to solve which is to have formatting for URLs. There are a few ways we can consider this. 

@tdilauro the displayFormat was added for compound fields, because that's where you were "grouping" things and that's really what it's for. We should actually be handing type URL better.

Something like software is more tricky because what if that's not used for a URL? In that case, we may want consider allowing html in that field, like we do for some others. (I *think* the rule is if it is a textbox you can do html and it is rendered - whitelisted, of course).

**Comment by tdilauro** [comment]
@scolapasta Sorry, I meant fields in our software metadata block, not the software field. There are other fields in this block, as well, but these are the link kinds of fields:

dvndb=# select name, title, fieldtype, displayformat from  datasetfieldtype where displayformat like '<a%' and name like 'sw%';
 swCodeRepositoryLink       | Code Repository Link            | URL       | <a href="#VALUE">#VALUE</a>
 swDependencyLink           | Dependency Location             | URL       | <a href="#VALUE">#VALUE</a>
 swInputOutputLink          | Input/Output Location           | URL       | <a href="#VALUE">#VALUE</a>
 swOtherRelatedSoftwareLink | Other Related software Location | URL       | <a href="#VALUE">#VALUE</a>
 swOtherRelatedResourceLink | Other Related resource Location | URL       | <a href="#VALUE">#VALUE</a>
(5 rows)

**Comment by edzale** [comment]
Hi, I am experiencing the same issue and after reading your comments, I am a bit confused: is it a bug or a matter of misconfiguration?

**Comment by scolapasta** [comment]
You are correct it is a bug. I closed it since it's already tracked by that
last issue.

**Comment by edzale** [comment]
Hi @jggautier ,
I changed the displayFormat to <a href="#VALUE">#VALUE</a> and it remains unclickable (https://data.inra.fr/dataset.xhtml?persistentId=doi:10.15454/1.505380010373349E12). If there is a bug here as stated by @scolapasta when do you think it will be handled? thanks in advance for your help.
Esther

**Comment by jggautier** [comment]
Hi @edzale,

You're right. This issue is unresolved. Changing the displayFormat of the primitive field "alternativeURL" to `<a href="#VALUE">#VALUE</a>` won't work because it works only for compound fields.

I defer to @djbrooke to help answer when it'll be handled.

(I edited your comment to make the displayFormat display as code `<a href="#VALUE">#VALUE</a>`. Otherwise it looks like #VALUE, which is another displayFormat and has caused confusion before :) .)

**Comment by edzale** [comment]
Hi @jggautier 
Thank you for your answer. Looking forward to a solution.
Esther

**Comment by pdurbin** [comment]
@edzale to be clear, you're talking specifically about the "Alternative URL" field, right? The screenshot below is from the dataset you mentioned and like you said, it isn't clickable:

![screen shot 2018-02-26 at 1 38 55 pm](https://user-images.githubusercontent.com/21006/36688481-99ce7302-1afa-11e8-9a57-a2aac7426cfe.png)

**Comment by edzale** [comment]
@pdurbin, that's right, I am talking about the "Alternative URL" field.

**Comment by umuchlish** [comment]
Hi @pdurbin 

Alternative URL has working perfectly at [my end](https://data.cifor.org/dataset.xhtml?persistentId=doi:10.17528/CIFOR/DATA.00049), by modify alternativeURL record in the datasetfieldtype table, add `<a href="#VALUE">#VALUE</a>` to displayFormat field.
 
![image](https://user-images.githubusercontent.com/906898/40289490-d2ca98aa-5ce2-11e8-8b36-01ad90b13487.png)

I saw [citation.tsv ](https://github.com/IQSS/dataverse/blob/b4e1f2a284f16ecd9ac1a9cfaac9e1393fd004ae/scripts/api/data/metadatablocks/citation.tsv#L7) has not included "displayFormat" with `<a href="#VALUE">#VALUE</a>` since last night upgrade to 4.8.6 not effecting to existing altearnativeURL.

**Comment by umuchlish** [comment]
Sure Philip.

**Comment by pdurbin** [comment]
@umuchlish thanks for pull request #4711! I just moved it to code review at https://waffle.io/IQSS/dataverse . It looks fine to me. Thanks!

**Comment by mheppler** [comment]
I was going to add this to a comment in the PR, but I wanted to float the suggestion here in the issue, in case there was reason not to follow this approach.

I have noticed that the `<a href="#VALUE">#VALUE</a>` for the alternativeURL does not include the attribute `target="_blank"` which opens the link in a new window. This is the approach we take with external links throughout the app. But apparently that is not the case in the `citation.tsv` file.

The links for citation fields alternativeURL, publicationURL, keywordVocabularyURI, topicClassVocabURI, producerURL, distributorURL, are all missing this attribute. I suggest that we add it for all of these and make sure to keep an eye of other links in the `.tsv` metadata blocks that are also missing it.

@scolapasta @sbarbosadataverse @jggautier Thoughts?

**Comment by pameyer** [comment]
Another thought that's almost certainly out of scope for this issue, but related to the discussion of consistency with handling external links, would be adding `rel="noopener"` to external links (recommended by lighthouse for performance / security - https://developers.google.com/web/tools/lighthouse/audits/noopener).

@mheppler @djbrooke (and anybody else interested) - think it's worth a new issue?

**Comment by mheppler** [comment]
I think that if we're adding one attribute to the links in this metadata block, then it perfectly in scope to also add a second, especially if it is a security AND performance improvement!

**Comment by djbrooke** [comment]
@umuchlish - thanks for the PR. Are you able and willing to make the change that @mheppler suggested above? 

@pameyer - can you please create a separate issue for your suggestion?

**Comment by umuchlish** [comment]
Hi djbrooke, ah yes, that would be better.

**Comment by pdurbin** [comment]
In 6ffb28d I added `target="_blank"` to all the fields @mheppler mentioned. Now they open in a new window. Here's how the look in the UI:

![screen shot 2018-05-29 at 9 02 37 am](https://user-images.githubusercontent.com/21006/40660364-c3bee45a-631e-11e8-96ef-c1735b5404a9.png)

Thank you again to @umuchlish for the pull request! The fix should also benefit @shlake and @edzale who have commented on this issue as well as others.

I moved this issue to QA at https://waffle.io/IQSS/dataverse

It's small so I'd love it if we could get this in 4.9.
